### PR TITLE
added vertical tab properties and alphabetized them

### PR DESCRIPTION
**Pull request by Blackbaud-DeniseCisneros**
Addresses https://github.com/blackbaud/skyux2/issues/1840

Would definitely appreciate the extra eyes to make sure I added these to the right spot 😬

### CODE REVIEW

File: src/app/components/vertical-tabs/index.html [hunk 1]
```diff
@@ -38,10 +66,10 @@
     Displays a counter alongside the tab header.
     </sky-demo-page-property>
     <sky-demo-page-property
-      propertyName="disabled"
+      propertyName="tabID"
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: The "d" should not be capitalized: "tabId"

File: src/app/components/vertical-tabs/index.html [hunk 2]
```diff
@@ -20,16 +20,44 @@
...
 
   <sky-demo-page-properties sectionHeading="Vertical tab properties">
+    <sky-demo-page-property
+      propertyName="tabHeading"
+    >
+    Specifies the tab header.
+    </sky-demo-page-property>
     <sky-demo-page-property
       propertyName="active"
       isOptional="true"
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: Add `defaultValue="multiple"` for this property.

File: src/app/components/vertical-tabs/index.html [hunk 3]
```diff
@@ -38,10 +66,10 @@
     Displays a counter alongside the tab header.
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: Add "This property accepts `number` values." to the end

File: src/app/components/vertical-tabs/index.html [hunk 4]
```diff
@@ -20,16 +20,44 @@
...
+    Specifies an ARIA role for the tab to support accessibility by indicating how the tab functions and what it controls. For information about ARIA roles, see the <a href="https://www.w3.org/WAI/PF/aria/roles">WAI-ARIA roles model</a>. This property accepts <stache-code>string</stache-code> values.
+    </sky-demo-page-property>
+    <sky-demo-page-property
+      propertyName="ariaRequired"
+      isOptional="true"
+    >
+    Specifies which values are required for the form to be complete when the tabset is part of a form. This property accepts <stache-code>boolean</stache-code> values.
+    <sky-demo-page-property
+      propertyName="disabled"
+      isOptional="true"
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: Add `defaultValue="false"` for this property.

File: src/app/components/vertical-tabs/index.html [hunk 5]
```diff
@@ -20,16 +20,44 @@
...
-    Indicates whether the tab is active when the tabset loads.
+    Indicates whether the tab is active when the tabset loads. This property accepts <stache-code>boolean</stache-code> values.
     </sky-demo-page-property>
     <sky-demo-page-property
-      propertyName="tabHeading"
+      propertyName="ariaControls"
+      isOptional="true"
     >
-    Specifies the tab header.
+    Specifies an ID to identify the element that contains the content the vertical tab exposes. This property supports <a href="https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure">accessibility rules for disclosures</a>. This property accepts <stache-code>string</stache-code> values.
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: Should we should call out that this corresponds to the value of the `tabId` property? It seems like that's what the notes indicate at https://github.com/blackbaud/skyux2/issues/1840. Am I misreading that?

File: src/app/components/vertical-tabs/index.html [hunk 6]
```diff
@@ -20,16 +20,44 @@
...
+    </sky-demo-page-property>
+    <sky-demo-page-property
+      propertyName="ariaInvalid"
+      isOptional="true"
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: Add `defaultValue="false"` for this property.

File: src/app/components/vertical-tabs/index.html [hunk 7]
```diff
@@ -20,16 +20,44 @@
...
+    Specifies which values are invalid in the tab's content when the tabset is part of a form. This property accepts <stache-code>boolean</stache-code> values.
+    <sky-demo-page-property
+      propertyName="ariaRole"
+      isOptional="true"
+    >
+    Specifies an ARIA role for the tab to support accessibility by indicating how the tab functions and what it controls. For information about ARIA roles, see the <a href="https://www.w3.org/WAI/PF/aria/roles">WAI-ARIA roles model</a>. This property accepts <stache-code>string</stache-code> values.
+    </sky-demo-page-property>
+    <sky-demo-page-property
+      propertyName="ariaRequired"
+      isOptional="true"
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: Add `defaultValue="false"` for this property.

File: src/app/components/vertical-tabs/index.html [hunk 8]
```diff
@@ -20,16 +20,44 @@
...
-    Specifies the tab header.
+    Specifies an ID to identify the element that contains the content the vertical tab exposes. This property supports <a href="https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure">accessibility rules for disclosures</a>. This property accepts <stache-code>string</stache-code> values.
+    </sky-demo-page-property>
+    <sky-demo-page-property
+      propertyName="ariaInvalid"
+      isOptional="true"
+    >
+    Specifies which values are invalid in the tab's content when the tabset is part of a form. This property accepts <stache-code>boolean</stache-code> values.
+    <sky-demo-page-property
+      propertyName="ariaRole"
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: `ariaRole` and `ariaRequired` need to be swapped for alphabetic order.

File: src/app/components/vertical-tabs/index.html [hunk 9]
```diff
@@ -20,16 +20,44 @@
...
+      isOptional="true"
     >
-    Specifies the tab header.
+    Specifies an ID to identify the element that contains the content the vertical tab exposes. This property supports <a href="https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure">accessibility rules for disclosures</a>. This property accepts <stache-code>string</stache-code> values.
+    </sky-demo-page-property>
+    <sky-demo-page-property
+      propertyName="ariaInvalid"
+      isOptional="true"
+    >
+    Specifies which values are invalid in the tab's content when the tabset is part of a form. This property accepts <stache-code>boolean</stache-code> values.
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: To follow the style pattern with boolean properties, change "Specifies which" to "Indicates whether." I'd also add "the" before "values." ... And based on the note in, I think I'd move "are invalid" after "content." I think it reads more naturally that way, and I think this wording changes the meaning slightly (although I might be mistaken about that).

File: src/app/components/vertical-tabs/index.html [hunk 10]
```diff
@@ -20,16 +20,44 @@
...
+      propertyName="ariaRole"
+      isOptional="true"
+    >
+    Specifies an ARIA role for the tab to support accessibility by indicating how the tab functions and what it controls. For information about ARIA roles, see the <a href="https://www.w3.org/WAI/PF/aria/roles">WAI-ARIA roles model</a>. This property accepts <stache-code>string</stache-code> values.
+    </sky-demo-page-property>
+    <sky-demo-page-property
+      propertyName="ariaRequired"
+      isOptional="true"
+    >
+    Specifies which values are required for the form to be complete when the tabset is part of a form. This property accepts <stache-code>boolean</stache-code> values.
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: To follow the style pattern with boolean properties, change "Specifies which" to "Indicates whether." I'd also add "the" before "values." ... And based on the note in https://github.com/blackbaud/skyux2/issues/1840, I think I'd add "in the tabset" after "values" and delete "for the form to be complete." I think it reads more naturally that way, and I think the angle about completing the form is implied by saying the fields are required.

File: src/app/components/vertical-tabs/index.html [hunk 11]
```diff
@@ -20,16 +20,44 @@
...
     <sky-demo-page-property
-      propertyName="tabHeading"
+      propertyName="ariaControls"
+      isOptional="true"
     >
-    Specifies the tab header.
+    Specifies an ID to identify the element that contains the content the vertical tab exposes. This property supports <a href="https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure">accessibility rules for disclosures</a>. This property accepts <stache-code>string</stache-code> values.
+    </sky-demo-page-property>
+    <sky-demo-page-property
+      propertyName="ariaInvalid"
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: I just noticed that the UX guidelines for vertical pages say "Do not use vertical tabs on a form. Use the sectioned form component instead." ... So the descriptions of `ariaInvalid` and `ariaRequired` contradict the UX guidelines. I think Conor provided the descriptions in https://github.com/blackbaud/skyux2/issues/1840 because he did the dev work, but I think we should reach out to Todd about this. It seems like we probably need to include a recommendation in the property descriptions to NOT use vertical tabs in forms, but maybe Todd will have a different recommendation.

Comment (Blackbaud-DeniseCisneros) [reply src/app/components/vertical-tabs/index.html]: I'll reach out to Todd about this one

Comment (Blackbaud-DeniseCisneros) [reply src/app/components/vertical-tabs/index.html]: Chatted with Alex, and he suggested moving `ariaInvalid` and `ariaRequired` to the Sectioned form component docs to stay consistent with design guidelines.

File: src/app/components/vertical-tabs/index.html (hunk 4 above)
Comment (blackbaud-johnly) [reply src/app/components/vertical-tabs/index.html]: This is marked as resolve, but I don't see the default value showing up when I serve the docs.

File: src/app/components/sectioned-form/index.html [hunk 12]
```diff
@@ -11,16 +11,30 @@
...
   <sky-demo-page-properties sectionHeading="Sectioned form section settings">
+    <sky-demo-page-property
+      propertyName="heading"
+    >
+    Specifies the section header.
+    </sky-demo-page-property>
     <sky-demo-page-property
       propertyName="active"
       isOptional="true"
     >
```
Comment (blackbaud-johnly) [inline src/app/components/sectioned-form/index.html]: Add `defaultValue="false"` for this property.

File: src/app/components/sectioned-form/index.html [hunk 13]
```diff
@@ -11,16 +11,30 @@
...
     Indicates whether the section is active when the form loads.
     </sky-demo-page-property>
     <sky-demo-page-property
-      propertyName="heading"
+      propertyName="ariaInvalid"
+      isOptional="true"
+      default="false"
```
Comment (blackbaud-johnly) [inline src/app/components/sectioned-form/index.html]: Change "default" to "defaultValue"

File: src/app/components/sectioned-form/index.html [hunk 14]
```diff
@@ -11,16 +11,30 @@
...
     >
-    Specifies the section header.
+    Indicates whether the values in the tab's content are invalid when the tabset is part of a form. This property accepts <stache-code>boolean</stache-code> values.
+    </sky-demo-page-property>
+    <sky-demo-page-property
+      propertyName="ariaRequired"
+      isOptional="true"
+      default="false"
```
Comment (blackbaud-johnly) [inline src/app/components/sectioned-form/index.html]: Change "default" to "defaultValue"

File: src/app/components/sectioned-form/index.html [hunk 15]
```diff
@@ -11,16 +11,30 @@
...
+    >
+    Indicates whether the values in the tabset are required when the tabset is part of a form. This property accepts <stache-code>boolean</stache-code> values.
     </sky-demo-page-property>
     <sky-demo-page-property
       propertyName="itemCount"
```
Comment (blackbaud-johnly) [inline src/app/components/sectioned-form/index.html]: In the description for this property, add "This property accepts `number` values."

File: src/app/components/sectioned-form/index.html [hunk 16]
```diff
@@ -11,16 +11,30 @@
   </sky-demo-page-summary>
 
   <sky-demo-page-properties sectionHeading="Sectioned form section settings">
+    <sky-demo-page-property
+      propertyName="heading"
+    >
+    Specifies the section header.
```
Comment (blackbaud-johnly) [inline src/app/components/sectioned-form/index.html]: In the description for this property, add "This property accepts `string` values."

File: src/app/components/sectioned-form/index.html [hunk 17]
```diff
@@ -11,16 +11,30 @@
...
+    </sky-demo-page-property>
     <sky-demo-page-property
       propertyName="active"
       isOptional="true"
     >
     Indicates whether the section is active when the form loads.
```
Comment (blackbaud-johnly) [inline src/app/components/sectioned-form/index.html]: In the description for this property, add "This property accepts `boolean` values."

File: src/app/components/vertical-tabs/index.html [hunk 18]
```diff
@@ -20,28 +20,47 @@
...
-    Indicates whether the tab is active when the tabset loads.
+    Indicates whether the tab is active when the tabset loads. This property accepts <stache-code>boolean</stache-code> values.
     </sky-demo-page-property>
     <sky-demo-page-property
-      propertyName="tabHeading"
+      propertyName="ariaControls"
+      isOptional="true"
     >
-    Specifies the tab header.
+    Specifies an ID that corresponds to the <stache-code>tabId</stache-code> and identifies the element that contains the content the vertical tab exposes. This property supports <a href="https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure">accessibility rules for disclosures</a>. This property accepts <stache-code>string</stache-code> values.
```
Comment (blackbaud-johnly) [inline src/app/components/vertical-tabs/index.html]: I think this first sentence is hard to follow. I know I recommended calling out that this value corresponds to the value of the `tabId` property, but I think it would be clearer if we make that info a second sentence:

"Specifies an ID to identify the element that contains the content that the vertical tab exposes. This ID corresponds to the <stache-code>tabId</stache-code>."

What do you think?

File: src/app/components/sectioned-form/index.html [hunk 19]
```diff
@@ -19,22 +19,37 @@
...
       propertyName="active"
       isOptional="true"
+      defaultValue="false"
     >
-    Indicates whether the section is active when the form loads.
+    Indicates whether the section is active when the form loads. This property accepts <stache-code>boolean</stache-code> values.
     </sky-demo-page-property>
     <sky-demo-page-property
-      propertyName="heading"
+      propertyName="ariaInvalid"
```
Comment (Blackbaud-AlexKingman) [inline src/app/components/sectioned-form/index.html]: I just noticed that while `ariaInvalid` and `ariaRequired` are properties of the vertical tabs, they aren't called the same thing in sectioned form sections. They're actually called `fieldInvalid` and `fieldRequired `. Who knows why?

Anyways, we'll probably need to update this to the latter above. I think your descriptions still hold true, because they just pass along the values to the tab's aria inputs. See here: https://github.com/blackbaud/skyux-tabs/blob/master/src/app/public/modules/sectioned-form/sectioned-form-section.component.html#L4

Comment (Blackbaud-DeniseCisneros) [reply src/app/components/sectioned-form/index.html]: Ah, thanks for the heads up! Fixing that now
